documentation/src/main/asciidoc/topics/ref_performance_tuning.adoc

```diff
@@ -79,6 +79,78 @@ The recommended values are:
 * send window size - 640 KB
 * receive window size - 25 MB
 
+== Thread pool settings
+
+A thread pool manages threads by assigning them concurrent tasks.
+Additionally, a thread pool can improve system performance by using any idle thread instead of creating threads.
+
+Thread pools limit the creation and deletion of threads to active worker threads, so that the amount of threads does not typically exceed the amount of tasks.
+This helps reduce the memory footprint of your {brandname} server.
+
+Your {brandname} configuration applies default settings to thread pools, but you can change these settings to meet your needs.
+
+The following table outlines common thread pool properties and their default settings:
+
+[cols="1,1"]
+|===
+| Property | Default setting | Description
+
+|`ASYNC_NOTIFICATION_EXECUTOR` |`1`| Used by asynchronous listeners that your {brandname} server registers in embedded mode. 
+The default setting ensures that Infinispan invokes asynchronous listeners according to importance of their messages. 
+Configuring this property is discouraged.
+Increasing the property's value from `1` might cause your {brandname} server to randomly create mixed types of listener events on different threads.
+|`BLOCKING_EXECUTOR` | `150` |  Used by your {brandname} server to handle blocking operations.
+An example of a blocking operation includes a cache store that calls external sources, transactions, or time-consuming non-blocking entities.
+Thread pool reuses blocking threads based on the value specified for this property.
+For workloads with large amounts of blocking threads, you set a higher value for the `BLOCKING_EXECUTOR` property.
+|`EXPIRATION_SCHEDULED_EXECUTOR`|`1`| Based on the property's value, a reaper schedules checks on expired cache entries that exist in memory.
+If your workload includes a large amount of expired cache entries that span multiple caches, consider setting a higher value for this property. 
+Be mindful that setting a higher value does not increase the rate at which threads process expired cache entries.
```

Comment:
Also if there is a single cache with huge expiration it will not help as one thread will handle one cache!

Reply:
I added:

> For example, a single cache with a high expiration value might have just one thread to handle all of the cache's entries.

Should I mention the possible latency issue with this use case or would this be assumed?

Reply:
I would mention it as the user might not assume that the the expiration scheduler will use one thread for one cache

Reply:
HI @wfink . I changed to:

> The default setting of the `EXPIRATION_SCHEDULED_EXECUTOR` property causes one thread to handle all cache entries for a single cache.